### PR TITLE
Update core system contracts for new e2e tests

### PR DESCRIPTION
**Pull request by piersy**
### Description

The tests under `./e2e_test` were running against the system
contracts at version `celo-core-contracts-v3.rc0`, this PR updates
that version to be `core-contracts.v6` (because that is the most recently deployed version) .

Please test this by running `make prepare-system-contracts` and checking that the monorepo checkout (normally under `../.celo-blockchain-monorepo-checkout` relative to the `celo-blockchain` root) is set to version `core-contracts.v6` and that the contracts under `./compiled-system-contracts` have been updated and then run the new e2e tests `go test ./e2e_test` and check that they pass.

You will need node v12 installed for this to work, which is a change from the previous requirement of v10.

### Other changes

The make rule `prepare-system-contracts` was broken and didn't
build new versions when the target version was changed. The mechanism
for setting the version has been changed, it is now stored in a file 
(`monorepo_commit`) at the repo root, and the make rule will now rebuild
the contracts if the version in this file is changed.

CI and the README have been updated accordingly.

### Backwards compatibility

This just changes how we run our tests.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit f3b73871124bace8f46362690ea798d82dbb280d
<details><summary>coverage:  51.2% of statements across all listed packages</summary><br>
<pre>
coverage:  61.2% of statements in consensus/istanbul
coverage:  42.2% of statements in consensus/istanbul/announce
coverage:  56.2% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  69.6% of statements in consensus/istanbul/core
coverage:  50.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  64.4% of statements in consensus/istanbul/uptime
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 99274f8a6a
</details>

File: README.md [hunk 1]
```diff
@@ -59,28 +59,29 @@ The Celo blockchain client comes with several wrappers/executables found in the
...
+subsequently edit the system contracts source or `monorepo_commit`, running the
+make rule again will re-checkout the monorepo, re-compile the contracts and
+copy them into place. 
 
 This make rule will shallow checkout
 [celo-monorepo](https://github.com/celo-org/celo-monorepo) under
-`../.celo-blockchain-monorepo-checkout` relative to this project's root, and it
-will checkout the commit defined in the variable MONOREPO_COMMIT in the
-Makefile. 
+`../.celo-blockchain-monorepo-checkout` relative to this project's root.
```
Comment (gastonponti) [inline README.md:74]: We should probably remove this second paragraph, right?

File: README.md [hunk 2]
```diff
@@ -59,28 +59,29 @@ The Celo blockchain client comes with several wrappers/executables found in the
...
 ```
-make prepare-system-contracts MONOREPO_COMMIT=master MONOREPO_PATH=../alt-monorepo
-```
+make prepare-system-contracts MONOREPO_PATH=../alt-monorepo
+``` 
 
 Without first running this make rule, certain tests will fail with errors such
 as:
 
-```
```
Comment (gastonponti) [inline README.md:83]: probably not intended

**Review by gastonponti — COMMENTED** [review]
This command works if the MONOREPO_PATH does not exist, or if it don't have the /build (I think)
I'm trying to run this with an old monorepo that I have, and it never checked out the v6 version

**Comment by piersy** [comment]
> This command works if the MONOREPO_PATH does not exist, or if it don't have the /build (I think) I'm trying to run this with an old monorepo that I have, and it never checked out the v6 version

Hey @gastonponti, it's not intended to work with a manually checked out monorepo. I've updated to README to try and make that clear, could you take a look here - https://github.com/celo-org/celo-blockchain/pull/1893/commits/42cc00dda0e97226e383d93429cc4016622fcdc4 ?

**Review by gastonponti — APPROVED** [review]
LGTM

**Pull request merged**